cc/mcts_node.cc

```diff
@@ -28,54 +28,37 @@ namespace minigo {
 
 namespace {
 
-void InitLegalMoves(MctsNode* node) {
-  auto& position = node->position;
-  auto position_hash = position.stone_hash();
-  auto to_play = position.to_play();
-  for (int c = 0; c < kN * kN; ++c) {
-    switch (position.ClassifyMove(c)) {
-      case Position::MoveType::kIllegal: {
-        // The move is trivially not legal.
-        node->legal_moves[c] = false;
-        break;
-      }
-
-      case Position::MoveType::kNoCapture: {
-        // The move will not capture any stones: we can calculate the new
-        // position's stone hash directly.
-        auto new_hash = position_hash ^ zobrist::MoveHash(c, to_play);
-        node->legal_moves[c] = !node->HasPositionBeenPlayedBefore(new_hash);
-        break;
-      }
+// Superko implementation that uses MctsNode::superko_cache.
```

Comment:
this is a weird noun for a classname; it's not clear to me what a superko as a noun is but w/e.  I don't have any better ideas, really; "ZobristHistory" ? "SuperkoDetector?"  Nah.

Reply:
`ZobristHistory` sgtm